client/src/register/Register.js

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import axios from 'axios';
+
+import "../App.css";
+class Register extends React.Component {
+  state = {
+    username: "",
+    password: "",
+    department: ""
+  };
+
+  handleInputChange = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const endpoint = "http://localhost:9000/api/auth/register";
+
+    axios
+      .post(endpoint, this.state)
+      .then(res => {
+        localStorage.setItem("jwt", res.data.token);
+        this.props.history.push("/login");
+      })
+      .catch(e => {
+        console.error(e);
+      });
+  };
+
+  render() {
+    return (
+      <>
+        <h2>Register</h2>
+        <form onSubmit={this.handleSubmit}>
+          <div>
+            <label htmlFor="username">Username</label>
+            &nbsp;
+            <input
+              name="username"
+              id="username"
+              value={this.state.username}
+              onChange={this.handleInputChange}
+              type="text"
+            />
+            <label htmlFor="password">Password</label>
+            &nbsp;
+            <input
+              name="password"
+              id="password"
+              value={this.state.password}
+              onChange={this.handleInputChange}
+              type="password"
+            />
+            <label htmlFor="department">Department</label>
+            &nbsp;
+            <input
+              name="department"
+              id="department"
+              value={this.state.department}
+              onChange={this.handleInputChange}
+              type="text"
+            />
```

Comment:
These inputs seem repetitive; do you think you could make a component for them? Or, for the sake of an example, a method:
```javascript
makeInput = (name, type = 'text') => (
  <>
    <label htmlFor={name}>{name[0].toUpperCase() + name.slice(1)}</label>
    &nbsp;
    <input
      name={name}
      id={name}
      value={this.state[name]}
      onChange={this.handleInputChange}
      type={type}
    />
  </>
);
```
and the usage:
```javascript
<form onSubmit={this.handleSubmit}>
  </div>
    {this.makeInput('username')}
    {this.makeInput('password', 'password')}
    {this.makeInput('department')}
  </div>

  <div>
    <button type="submit">Register</button>
  </div>
</form>
```

Reply:
Try to apply this sort of thing for your Login component as well; you might even be able to just create a singular component for both Register and Login (it'd just be a matter of passing their respective props at that point, or something similar).